### DISCUSSION TITLE
build error on devel branch

### DISCUSSION THREAD

**Discussion by gnahraf** [dbeaver/dbeaver#14642] [Q&A]
Greetings,

I'm trying to build the `devel` branch (from console) and am getting a dependency error.

My environment:

>$ java -version
java version "15.0.2" 2021-01-19
Java(TM) SE Runtime Environment (build 15.0.2+7-27)
Java HotSpot(TM) 64-Bit Server VM (build 15.0.2+7-27, mixed mode, sharing)
$ javac -version
javac 15.0.2
$

The error:

>$ mvn clean package
>. .
[WARNING] No system packages found in profile nor toolchain for JavaSE-15, using current JRE system packages.
This can cause faulty depnedency resolution, consider adding a definition for a 'jdk' with id=JavaSE-15 in your toolchains.xml
>. .
>[INFO] Computing target platform for MavenProject: org.jkiss.dbeaver:org.jkiss.dbeaver.model:1.0.159-SNAPSHOT @ /Users/babak/git/dbeaver/plugins/org.jkiss.dbeaver.model/pom.xml
[INFO] Resolving dependencies of MavenProject: org.jkiss.dbeaver:org.jkiss.dbeaver.model:1.0.159-SNAPSHOT @ /Users/babak/git/dbeaver/plugins/org.jkiss.dbeaver.model/pom.xml
[INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=x86_64}
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.jkiss.dbeaver.model 1.0.159.qualifier
[ERROR]   Missing requirement: org.jkiss.dbeaver.model 1.0.159.qualifier requires 'osgi.bundle; org.eclipse.equinox.security 0.0.0' but it could not be found
[ERROR] 
[ERROR] See https://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting for help.

This is my first try. I did consult the above troubleshooting page, but couldn't make much of it. (On googling, I found a similar error from years ago, related to an upstream maven dep, but nothing recent.) I'm not sure what that `[WARNING]` says, but seems prescient given what follows ;)

~Babak

**Comment by gnahraf** [comment]
Also, per the above warning, I created a `toolchains.xml` file in my `.m2` subdirectory

```
$ cat toolchains.xml
<toolchains>
  <toolchain>
     <type>jdk</type>
     <provides>
       <id>JavaSE-15</id>
     </provides>
     <configuration>
       <jdkHome>/Library/Java/JavaVirtualMachines/jdk-15.0.2.jdk/Contents/Home</jdkHome>
     </configuration>
  </toolchain>
 </toolchains>
```

Still no cigar: same error message.

**Reply by ShadelessFox** [reply]
Hello @gnahraf,

You need to use Java 11.

**Reply by gnahraf** [reply]
Thanks for suggesting this. So I temporarily switched jdks like this
>
>$ export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk-11.0.7.jdk/Contents/Home
$ export PATH=$JAVA_HOME/bin:$PATH
$ java -version
java version "11.0.7" 2020-04-14 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.7+8-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.7+8-LTS, mixed mode)
$ javac -version
javac 11.0.7

and updated `~/.m2/toolchains.xml` so it now reads

`<toolchains>
  <toolchain>
     <type>jdk</type>
     <provides>
       <id>JavaSE-11</id>
     </provides>
     <configuration>
       <jdkHome>/Library/Java/JavaVirtualMachines/jdk-11.0.7.jdk/Contents/Home</jdkHome>
     </configuration>
  </toolchain>
 </toolchains>
`

but still no luck:

>
>$ mvn clean
>[INFO] Scanning for projects...
[INFO] Computing target platform for MavenProject: org.jkiss.dbeaver:org.jkiss.utils:2.1.153-SNAPSHOT @ /Users/babak/git/dbeaver/bundles/org.jkiss.utils/pom.xml
[INFO] Resolving dependencies of MavenProject: org.jkiss.dbeaver:org.jkiss.utils:2.1.153-SNAPSHOT @ /Users/babak/git/dbeaver/bundles/org.jkiss.utils/pom.xml
[INFO] Resolving class path of MavenProject: org.jkiss.dbeaver:org.jkiss.utils:2.1.153-SNAPSHOT @ /Users/babak/git/dbeaver/bundles/org.jkiss.utils/pom.xml
[INFO] Computing target platform for MavenProject: org.jkiss.dbeaver:org.jkiss.wmi:2.0.163-SNAPSHOT @ /Users/babak/git/dbeaver/bundles/org.jkiss.wmi/pom.xml
[INFO] Resolving dependencies of MavenProject: org.jkiss.dbeaver:org.jkiss.wmi:2.0.163-SNAPSHOT @ /Users/babak/git/dbeaver/bundles/org.jkiss.wmi/pom.xml
[INFO] Resolving class path of MavenProject: org.jkiss.dbeaver:org.jkiss.wmi:2.0.163-SNAPSHOT @ /Users/babak/git/dbeaver/bundles/org.jkiss.wmi/pom.xml
[INFO] Computing target platform for MavenProject: org.jkiss.dbeaver:org.jkiss.dbeaver.model:1.0.159-SNAPSHOT @ /Users/babak/git/dbeaver/plugins/org.jkiss.dbeaver.model/pom.xml
[INFO] Resolving dependencies of MavenProject: org.jkiss.dbeaver:org.jkiss.dbeaver.model:1.0.159-SNAPSHOT @ /Users/babak/git/dbeaver/plugins/org.jkiss.dbeaver.model/pom.xml
[INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=x86_64}
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.jkiss.dbeaver.model 1.0.159.qualifier
[ERROR]   Missing requirement: org.jkiss.dbeaver.model 1.0.159.qualifier requires 'osgi.bundle; org.eclipse.equinox.security 0.0.0' but it could not be found
